tracks/typescript/exercises/resister-color-duo/mentoring.md

```diff
@@ -0,0 +1,47 @@
+
+### Reasonable Solutions
+
+```typescript
+export class ResistorColor {
+  
+  private readonly Colors: string[] = ["black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"]
+  
+  private readonly MINIMUM_ACCEPTABLE_COLORS = 2;
+
+  constructor(private colors: string[]) {
+    
+    if (colors.length < this.MINIMUM_ACCEPTABLE_COLORS)
+      throw "At least two colors need to be present";
+
+  }
+
+  value = (): number => {
+    
+    const result = this.colors
+      .slice(0, this.MINIMUM_ACCEPTABLE_COLORS)
+      .reduce(
+        (accumulator: string, current: string) =>
+          `${accumulator}${this.Colors.indexOf(current)}`,
+          ''
+      );
+
+    return parseInt(result);
```

Comment:
I've noticed that across the JS track at least we seem to prefer `Number()` over `parseX`, with the only reasoning I've ever found being that "it's more declarative". Thoughts on this? Are both acceptable?

Reply:
True, I think it is just that Number is faster?

Anyway, I have changed the example code to use Number now and added a bullet point about this to the list of suggestions. 👍

Reply:
I'm not entirely sure which is faster to be honest - and on exercism I think generally performance is not something that comes up too often in reviews. I typically refer people to this [SO Post](https://stackoverflow.com/a/4090554/9758007) that goes into more detail!